### DISCUSSION TITLE
How to parse a non-standard time unit format as years since 850-01-01 0:0:0 in xarray library for windows？

### DISCUSSION THREAD

**Discussion by KervinWen** [pydata/xarray#7610] [Q&A] [answered]
Recently, I have been working with some netCDF data that has a time dimension in the format of years since 850-01-01 0:0:0. However, I encountered some issues when using xarray to read in and analyze the data.
Specifically, I found that xarray is not able to parse this time format directly, and I also tried using other libraries like netcdf4 and cftime, but to no avail.

ValueError: In general, units must be one of 'microseconds', 'milliseconds', 'seconds', 'minutes', 'hours', or 'days' (or select abbreviated versions of these).   For the '360_day' calendar, 'months' can also be used, or for the 'noleap' calendar 'common_years' can also be used. Got 'years' instead, which are not recognized.

I understand that xarray has been designed to handle various types of time formats, but I have not been able to find a solution for this particular format.
Thank you very much for your time and consideration.

**Comment by spencerkclark** [comment, marked as answer]
Through underlying libraries (cftime in this case) xarray can typically understand date formats like the one listed when decoding times, but the main issue is the use of `"years"` as the time unit.  `"years"` can have an ambiguous meaning, so the common practice is to not make any assumptions.  This paragraph in the [Time Coordinate section of the CF Conventions](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.10/cf-conventions.html#time-coordinate) provides some context:

> UDUNITS defines a `year` to be exactly 365.242198781 days (the interval between 2 successive passages of the sun through vernal equinox). *It is not a calendar year*. UDUNITS defines a `month` to be exactly `year/12`, which is *not a calendar month*. The CF standard follows UDUNITS in the definition of units, but we recommend that `year` and `month` should not be used, because of the potential for mistakes and confusion.

There are ways to work around this manually, however.  What kind of calendar does your data use and what is the convention used for the meaning of the `"years"` units in your case?

**Reply by KervinWen** [reply]
I know that. Here some infomation about  `time` variable：

```
Variable name: time
axis: T
bounds: bounds_time
calendar: noleap
long_name: time
standard_name: time
units: years since 850-01-01 0:0:0
```

**Reply by spencerkclark** [reply]
Thanks -- the following should work in your case:
```python
DAYS_PER_YEAR_NOLEAP = 365
ds = xr.open_dataset("placeholder.nc", decode_times=False)
with xr.set_options(keep_attrs=True):
    ds["time"] = DAYS_PER_YEAR_NOLEAP * ds.time
ds["time"].attrs["units"] = ds["time"].attrs["units"].replace("years", "days")
ds = xr.decode_cf(ds)
```
You may need to do the same thing for the `"bounds_time"` variable, or simply drop it before the `decode_cf` step if you are not interested in using it.

**Reply by KervinWen** [reply]
Thank you very much. 
Good morning,maybe. Have a nice day~

**Reply by spencerkclark** [reply]
Thanks, same to you!
